### PR TITLE
Support for BareMetal and Hor.Autoscale. Added timeout to WaitUntilComplete

### PR DESCRIPTION
**Pull request by EfrenRey**
Hi, I'm Efren from ElasticBox team (CenturyLink).

We had included in ElasticBox support for BareMetal servers and Horizontal Autoscaling, and we would like include that functionality in the CLC Python library.

This PR include the following changes:
- Added `BareMetalCapabilities` method to `Datacenter` class.
- Added some methods in `Group` class to retrieve the Horizontal Autoscaling policy currently applied, apply a new one, or remove existing applied.
- Added a new `HorizontalAutoscalePolicy` class, with methods to list, create, delete, update, apply to group and remove from group.
- Added an optional timeout parameter on `WaitUntilComplete` method, because in case of a failure in CLC the loop would never ends.
- Updated `Create` method on `Server` class to support BareMetal servers creation.

Thanks.

### CODE REVIEW

**Comment by ack** [comment]
Efren - 

Thanks for the patches! I like that we're adding new functionality and this module is getting some love. In general however, despite any disagreements one might have with the style of the module, patches shouldn't wildly deviate from the existing code style. This pertains to everything from whitespace conventions to instance-vs-static-method structure. No matter how ugly it may seem, the code should read consistently where possible. 

Will leave some concrete notes in inline reviews.

File: src/clc/APIv2/server.py [hunk 1]
```diff
@@ -594,16 +595,40 @@ def Create(name,template,group_id,network_id,cpu=None,memory=None,alias=None,pas
...
-		         json.dumps({'name': name, 'description': description, 'groupId': group_id, 'sourceServerId': template,
-							 'isManagedOS': managed_os, 'primaryDNS': primary_dns, 'secondaryDNS': secondary_dns,
-							 'networkId': network_id, 'ipAddress': ip_address, 'password': password,
-							 'sourceServerPassword': source_server_password, 'cpu': cpu, 'cpuAutoscalePolicyId': cpu_autoscale_policy_id,
-							 'memoryGB': memory, 'type': type, 'storageType': storage_type, 'antiAffinityPolicyId': anti_affinity_policy_id,
-							 'customFields': custom_fields, 'additionalDisks': additional_disks, 'ttl': ttl, 'packages': packages}),
-				 		 session=session),
-				 alias=alias,
-				 session=session))
+		if type == 'bareMetal':
```
Comment (ack) [inline src/clc/APIv2/server.py]: prepare a base dictionary, and when 'bareMetal', add some additional fields. 

```python
payload = dict(name=name, description=description, ...)

if type.lower() == 'baremetal': 
    payload['configurationId'] = configuration_id
    ...

return clc.v2.Requests(clc.v2.API.Call('POST', 'servers/%s' % (alias), json.dumps(payloads), alias=alias, session=session)

```

File: src/clc/APIv2/queue.py [hunk 2]
```diff
@@ -141,6 +142,9 @@ def WaitUntilComplete(self,poll_freq=2):
 				elif status in ("failed", "unknown"): self.error_requests.append(request)
 
 			self.requests = cur_requests
+			if self.requests > 0 and timeout and (time.time() - start_time) > timeout:
```
Comment (ack) [inline src/clc/APIv2/queue.py]: consider propagating the request status in a new typed exception and/or pulling out this duplicate code into a function call. the request status might reveal that the condition is not a failure state for instance. refactoring as a re-used function call would reduce code.

File: src/clc/APIv2/server.py (hunk 1 above)
Comment (EfrenRey) [inline src/clc/APIv2/server.py]: @ack Agree. Done the change.

File: src/clc/APIv2/queue.py (hunk 2 above)
Comment (EfrenRey) [inline src/clc/APIv2/queue.py]: @ack Created a new ```RequestTimeoutException``` class with the current request status.
Also added a function ```clc.v2.time_utils.TimeoutExpired```.

**Pull request merged**